rustls/src/tls13/hs.rs

```diff
@@ -0,0 +1,125 @@
+use crate::hash_hs::HandshakeHash;
+use crate::msgs::base::Payload;
+use crate::msgs::enums::{AlertDescription, HandshakeType};
+use crate::msgs::handshake::{
+    CertificateEntry, CertificateExtension, CertificatePayloadTLS13, CertificateStatus,
+    DigitallySignedStruct, HandshakeMessagePayload, HandshakePayload,
+};
+use crate::msgs::message::{Message, MessagePayload};
+use crate::sign::Signer;
+use crate::{verify, Certificate, CommonState, Error, ProtocolVersion};
+use log::{trace, warn};
+use ring::{constant_time, hmac};
+
+pub(crate) fn emit_certificate_and_certificate_verify(
+    transcript: &mut HandshakeHash,
+    common: &mut CommonState,
+    context: Option<Vec<u8>>,
+    cert_chain: Vec<Certificate>,
+    ocsp_response: Option<&[u8]>,
+    sct_list: Option<&[u8]>,
+    signer: Option<&dyn Signer>,
+    context_string_with_0: verify::ContextString,
+) -> Result<(), Error> {
+    let signer = signer.ok_or_else(|| {
+        common.send_fatal_alert(AlertDescription::HandshakeFailure);
+        Error::PeerIncompatibleError("no overlapping sigschemes".to_string())
```

Comment:
At least one test is failing because of this, as this is a change in behavior for the TLS 1.3 client, but I think that test may be relying on buggy behavior. I filed PR #969 about it.